akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala

```diff
@@ -402,6 +420,16 @@ private[transport] class ProtocolStateActor(initialData: InitialProtocolStateDat
 
       }
 
+    case Event(HandshakeTimer, OutboundUnderlyingAssociated(_, wrappedHandle)) ⇒
+      sendDisassociate(wrappedHandle, Unknown)
+      stop(FSM.Failure(TimeoutReason("No response from remote for outbound association. Handshake timed out after " +
+        s"[${settings.HandshakeTimeout.toMillis} ms].")))
+
+    case Event(HandshakeTimer, InboundUnassociated(_, wrappedHandle)) ⇒
+      sendDisassociate(wrappedHandle, Unknown)
+      stop(FSM.Failure(TimeoutReason("No response from remote for inbound association. Handshake timed out after " +
+        s"[${settings.HandshakeTimeout.toMillis} ms].")))
```

Comment:
I'm unsure about the timeout for inbound associations. It was not handled previously, and is perhaps not needed, but it might not hurt either?